libs/langchain-community/src/vectorstores/pgvector.ts

```diff
@@ -163,17 +163,19 @@ export class PGVectorStore extends VectorStore {
    * `connect` to return a new instance of `PGVectorStore`.
    *
    * @param embeddings - Embeddings instance.
-   * @param fields - `PGVectorStoreArgs` instance.
+   * @param fields - `PGVectorStoreArgs` instance
+   * @param fields.dimensions Number of dimensions in your vector data type. For example, use 1536 for OpenAI's `text-embedding-3-small`. If not set, indexes like HNSW might not be used during query time.
    * @returns A new instance of `PGVectorStore`.
    */
   static async initialize(
     embeddings: EmbeddingsInterface,
-    config: PGVectorStoreArgs
+    config: PGVectorStoreArgs & { dimensions?: number }
```

Comment:
Would it make sense to add the optional dimensions to the PGVectorStoreArgs type directly? 🤔

Reply:
Yeah I can now see how it's not convenient - I can just make the constructor `Omit` dimensions